app/components/Comments/CommentView.js

```diff
@@ -32,10 +32,29 @@ const CommentView = (props: Props) => {
     loggedIn,
     style,
     displayTitle = true,
-    newOnTop = false
+    newOnTop = false,
+    deleteComment
   } = props;
   const commentFormProps = { commentTarget, user, loggedIn };
-  const tree = generateTreeStructure(comments);
+
+  const commentIds = comments.reduce(
+    (obj, comment) => ({ ...obj, [comment.id]: true }),
+    { null: true }
+  );
+  const undefinedParents = new Set(
+    comments
+      .filter(comment => !commentIds[comment.parent])
+      .map(comment => comment.parent)
+  );
+
+  const emptyComments = [...undefinedParents].map(id => ({
```

Comment:
Maybe this should be implemented in the backend instead.. If the deleted parent was a reply, you lose that tree structure, since all emptyComments have no parent, even if the original comment did.
Thoughts?

Reply:
Yeah, the only issue is when you have a tree with three branches and the middle one is deleted. You could probably implement it in backend, but since the object is just flagged as deleted, you would have to update the keys of the related comments. Also, there is the issue with the user field being empty/null. So I'm not sure how to handle this, but I can take a look!